level1c4pps/isccpng2pps_lib.py

```diff
@@ -0,0 +1,268 @@
+                        BANDNAMES,
+                        PPS_TAGNAMES,
+                        chunks=None)
+
+
+def set_header_and_band_attrs(scene, orbit_n=00000):
+    """Set and delete some attributes."""
+    nimg = 0  # name of first dataset is image0
+    # Set some header attributes:
+    irch = scene['temp_11_00um']
+    irch.attrs['instrument'] = "seviri"
+    scene.attrs['source'] = "isccpng2pps.py"
+    scene.attrs['platform_name'] = "meteosat11"
+    nimg = set_header_and_band_attrs_defaults(scene, BANDNAMES, PPS_TAGNAMES, REFL_BANDS, irch, orbit_n=orbit_n)
+    for band in REFL_BANDS:
+        if band not in scene:
+            continue
+        scene[band].attrs['sun_zenith_angle_correction_applied'] = 'False'
+    return nimg
+
+
+def homogenize_channel(scene, wmo_id, illum, band, sol_zen):
+    """Homogenize one satellite for one band."""
+    data = scene[band]
+    dwmo_id = scene["wmo_id"].values
+    ch_index = coef_slope_chan.index(band)
+    k = coef_slope_list[str(wmo_id) + illum][ch_index]
+    c = coef_offset_list[str(wmo_id) + illum][ch_index]
+    logger.info(
+        f"Homogenizing channel {band} for {satellite_names[wmo_id]} for illum:{illum}"
+        f" with respect to SEVIRI on MSG4 using y={k} * x + {c}")
+    dn_discr = 90
+    if (illum == '_day'):
+        update = ((dwmo_id == wmo_id) & (data > 0) & (sol_zen < dn_discr))
+        scene[band].values = np.where(update, scene[band].values * k + c, scene[band].values)
+
+    if (illum == '_nig'):
+        update = ((dwmo_id == wmo_id) & (data > 0) & (sol_zen >= dn_discr))
+        scene[band].values = np.where(update, scene[band].values * k + c, scene[band].values)
+
+
+def homogenize(scene):
+    """Homogenize data to Meteosat-11."""
+    sol_zen = scene["sunzenith"]
+    for band in BANDNAMES:
+        for wmo_id in [270, 271, 173, 55]:
+            homogenize_channel(scene, wmo_id, '_day', band, sol_zen)
+            homogenize_channel(scene, wmo_id, '_nig', band, sol_zen)
+
+def recalibrate_meteosat(scene):
+    """Nominal calibration is applied, redo with meirnik calibration."""
+    from satpy.readers.seviri_base import MeirinkCalibrationHandler
+    start_time = dt64_to_datetime(scene["refl_00_65um"].attrs["start_time"])
+    channel_name = {"refl_00_65um": "VIS006",
+                    "refl_00_86um": "VIS008",
+                    "refl_01_60um": "IR_016"}
+    platform_id = {55: 321,
+                   70: 324}
+    # Normally read from file (HRIT). We do not have access to nominal calibration
+    calibration_nominal = {2021: {321: {"VIS006": 24.974,
+                                        "VIS008": 32.377,
+                                        "IR_016": 23.710},
+                                  324: {"VIS006": 21.241,
+                                        "VIS008": 27.921,
+                                        "IR_016": 23.112}
+                                  }}
+    for wmo_id in platform_id:
+        for band in channel_name:
```

Comment:
you could do: `for wmo_id, platform_id in PLATFORM_ID.items():` and  `for band, channel_name in CHANNEL_NAME.items():` to increase readability if you follow the advise above.